app/models.py

```diff
@@ -0,0 +1,607 @@
+        """sets the value of the url_name attribute"""
+        self.url_name = new_url_name
+
+    @property
+    def get_id(self):
+        """returns the id of a url object"""
+        return self.id
+
+    @staticmethod
+    def check_validity(new_url):
+        """
+        checks if the url has a valid url format
+        :param new_url: 
+        """
+        if not url(new_url):
+            raise UrlValidationException("Invalid url (Either url is empty"
+                                         " or of invalid format. (Url must include"
+                                         " either http:// or https://))")
+
+    @staticmethod
+    def get_from_json(json_data):
+        """
+        this function creates a Url object from the json data
+        passed in as argument
+        :param json_data: 
+        :return Url Object: 
+        """
+        return [
+            Url(url_name=json_data['url']),
+            json_data["vanity_string"]
+            if "vanity_string" in json_data else None,
+            json_data["shorten_url_length"]
+            if "shorten_url_length"in json_data else None
+        ]
+
+    @staticmethod
+    def get_url_by_name(url_name):
+        """
+        queries the database for a long url whose url_name 
+        matches the argument passed into the function 
+        :param url_name: 
+        :return Url object: 
+        """
+        return Url.query.filter_by(url_name=url_name).first()
+
+    @staticmethod
+    def get_shorten_url(new_url, vanity_string, short_url_length):
+        """
+        generates a shorten url for the url passed it as argument
+        but if a vanity string is passed, it uses the vanity_string instead
+        :param new_url: 
+        :param vanity_string: 
+        :param short_url_length: 
+        :return list contain ShortenUrl Object and a response message: 
+        """
+        existing_long_url = Url.get_url_by_name(new_url.name)
+        if existing_long_url and existing_long_url in g.current_user.url:
+            return ["Shorten Url already exist for this long url",
+                    list(set(existing_long_url.short_url)
+                         .intersection(g.current_user.short_url))[0]
+                    ]
+        shorten_url_name = Shortener.generate_shorten_name(short_url_length) \
+            if not vanity_string else vanity_string
+        while ShortenUrl.get_short_url_by_name(shorten_url_name):
+            shorten_url_name = Shortener\
+                                .generate_shorten_name(short_url_length)
+        if existing_long_url and not(existing_long_url in g.current_user.url):
+            Url.save(existing_long_url, shorten_url_name)
+        else:
+            Url.save(new_url, shorten_url_name)
+        return ["Url successful shoretened",
+                ShortenUrl.get_short_url_by_name(shorten_url_name)]
+
+    @staticmethod
+    def save(url, shorten_url_name):
+        """this function saves a long url to the database"""
```

Comment:
I have replaced the parameter name `url` in order to avoid it from shadowing the imported `url`.  Also a long url was saved in this method to the database using the `orm AppendBaseQuery` which was represented by `g.current_user.url.append(long_url)`